gateleen-core/src/main/java/org/swisspush/gateleen/core/util/HttpHeaderUtil.java

```diff
@@ -83,30 +78,28 @@ public static <T extends MultiMap> boolean hasMatchingHeader(@Nonnull T headers,
     }
 
     /**
-     * Merges headers, makes sure that only one value of header {@link HttpRequestHeader#CONTENT_LENGTH} ends up in the result.
+     * Merges headers, makes sure that only one value of all headers ends up in the destination MultiMap.
+     *
+     * Note: This is not 100% in line with https://www.rfc-editor.org/rfc/rfc7230#section-3.2.2 to be further looked into.
```

Comment:
I'm not 100% convinced yet and hope to find the root cause why we actually end up in situations where we have header duplication. However, we are currently blocked in our project due to this issue.

Reply:
Is there a specific pattern or header which is duplicated? Some expamples? This might give a sign to the root of the problem.

Reply:
<img width="444" alt="image" src="https://user-images.githubusercontent.com/30789476/165019450-f738c7df-65fa-4f3a-8ec7-e83441471f6b.png">